nixos/modules/services/networking/nebula.nix

```diff
@@ -227,9 +256,9 @@ in
           };
         }) enabledNetworks);
 
-    # Open the chosen ports for UDP.
-    networking.firewall.allowedUDPPorts =
-      unique (mapAttrsToList (netName: netCfg: netCfg.listen.port) enabledNetworks);
+    # Open the chosen ports for UDP, as well as the dns port if serving dns
+    networking.firewall.allowedUDPPorts = unique ((mapAttrsToList (netName: netCfg: netCfg.listen.port) enabledNetworks) ++
+      (mapAttrsToList(netName: netCfg: netCfg.dns.port)) enabledNetworks);
```

Comment:
is there a better way to do this? I was trying to fit a `lists.optional` in there to only open the DNS port when the dns is enabled, but wasn't sure where to add it. I am relatively new to the nixlang so appreciate any guidance here :smiley:

Reply:
This will conflict with https://github.com/NixOS/nixpkgs/pull/280836 FYI. And this is a fine way to do it, though I would add a filter for networks with DNS enabled.